add named connections to mssql (#2957)
grafana/alloy: docs/sources/reference/components/prometheus/prometheus.exporter.mssql.md

| `connection_string`    | `secret`   | The connection string used to connect to an Microsoft SQL Server.   |         | yes      |
| `connection_name`      | `string`   | The connection string used to connect to an Microsoft SQL Server.   |         | no       |
| `max_idle_connections` | `int`      | Maximum number of idle connections to any one target.               | `3`     | no       |
| `max_open_connections` | `int`      | Maximum number of open connections to any one target.               | `3`     | no       |
| `timeout`              | `duration` | The query timeout in seconds.                                       | `"10s"` | no       |
| `query_config`         | `string`   | MSSQL query to Prometheus metric configuration as an inline string. |         | no       |

The [`sql_exporter` examples](https://github.com/burningalchemist/sql_exporter/blob/master/examples/azure-sql-mi/sql_exporter.yml#L21) show the format of the `connection_string` argument:

```text
sqlserver://<USERNAME>:<PASSWORD>@<SQLMI_ENDPOINT>.database.windows.net:1433?encrypt=true&hostNameInCertificate=%2A.<SQL_MI_DOMAIN>.database.windows.net&trustservercertificate=true
```

> The `connection_name` parameter allows uptime metrics. Refer to the [`sql_exporter`](https://github.com/burningalchemist/sql_exporter#configuration) `target.name` setting.
> 

Reviewer: ```suggestion
The `connection_name` parameter allows uptime metrics.
Refer to the [`sql_exporter`](https://github.com/burningalchemist/sql_exporter#configuration) `target.name` setting.
```

Removed an extra CR/LF. Added semantic line break.

| Name                   | Type       | Description                                                         | Default | Required |
| ---------------------- | ---------- | ------------------------------------------------------------------- | ------- | -------- |
| `connection_string`    | `secret`   | The connection string used to connect to an Microsoft SQL Server.   |         | yes      |

> | `connection_name`      | `string`   | The connection string used to connect to an Microsoft SQL Server.   |         | no       |

Reviewer 2: This description looks to be the same as `connection_string`, and if it is *not* required, you should specify what the default value is.

Reviewer 2: Yes, "" would be correct.

> | `connection_name`      | `string`   | The name for the connection to report `up` metrics for.             | `""`    | no       |

Reviewer 2: Our docs writer won't forgive me if I approve a trailing preposition 🙃 .

```suggestion
| `connection_name`      | `string`   | The name of the connection, used as a label in uptime metrics.       | `""`    | no       |
```

| Name                   | Type       | Description                                                         | Default | Required |
| ---------------------- | ---------- | ------------------------------------------------------------------- | ------- | -------- |
| `connection_string`    | `secret`   | The connection string used to connect to an Microsoft SQL Server.   |         | yes      |

> | `connection_name`      | `string`   | The name for the connection to report `up` metrics for.             |         | no       |

Reviewer 2: Our docs writer won't forgive me if I approve a trailing preposition 🙃 .

> The name of the connection, used as a label in uptime metrics.

sqlserver://<USERNAME>:<PASSWORD>@<SQLMI_ENDPOINT>.database.windows.net:1433?encrypt=true&hostNameInCertificate=%2A.<SQL_MI_DOMAIN>.database.windows.net&trustservercertificate=true
```

The `connection_name` parameter allows uptime metrics.
Refer to the [`sql_exporter`](https://github.com/burningalchemist/sql_exporter#configuration) `target.name` setting.

> ```text

Reviewer 2: Why is there a copy of the connection string format here?
